### PR TITLE
Extract RunDeploy to be re-used elsewhere 👼

### PR DESCRIPTION
**Pull request by vdemeester**
Mainly useful to re-use in `docker/app` :angel: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by thaJeztah** [comment]
Ah! Needs a godoc

File: cli/command/stack/deploy.go [hunk 1]
```diff
@@ -19,20 +20,11 @@ func newDeployCommand(dockerCli command.Cli, common *commonOptions) *cobra.Comma
...
-			case common.orchestrator.HasKubernetes():
-				kli, err := kubernetes.WrapCli(dockerCli, kubernetes.NewOptions(cmd.Flags(), common.orchestrator))
-				if err != nil {
-					return err
-				}
-				return kubernetes.RunDeploy(kli, opts)
-			default:
-				return swarm.RunDeploy(dockerCli, opts)
+			commonOrchestrator := common.orchestrator
+			if common == nil { // Top-level deploy command
```
Comment (silvin-lubecki) [inline cli/command/stack/deploy.go]: You dereference `common` pointer just the line above 😅

Comment (mat007) [reply cli/command/stack/deploy.go]: and assign to it the line below 🤔

File: cli/command/stack/deploy.go [hunk 2]
```diff
@@ -19,20 +20,13 @@ func newDeployCommand(dockerCli command.Cli, common *commonOptions) *cobra.Comma
...
-				kli, err := kubernetes.WrapCli(dockerCli, kubernetes.NewOptions(cmd.Flags(), common.orchestrator))
-				if err != nil {
-					return err
-				}
-				return kubernetes.RunDeploy(kli, opts)
-			default:
-				return swarm.RunDeploy(dockerCli, opts)
+			var commonOrchestrator command.Orchestrator
+			if common == nil { // Top-level deploy command
+				common.orchestrator = command.OrchestratorSwarm
```
Comment (mat007) [inline cli/command/stack/deploy.go]: Shouldn't that be
```
common = &commonOptions{orchestrator: command.OrchestratorSwarm}
```
?

**Comment by vdemeester** [comment]
@thaJeztah @silvin-lubecki wait a tiny bit before merging :angel: I may have another slight change to do :stuck_out_tongue_closed_eyes:

**Comment by vdemeester** [comment]
Well, it's not the right change to do… let me rework something else :sweat_smile: 
Sorry :bowing_man:

**Pull request closed without merging**